corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -1146,28 +1150,31 @@ def get_ledger_value(case_id, section_id, entry_id):
             raise LedgerValueNotFound
 
     @staticmethod
-    def save_ledger_values(ledger_values, deprecated_form=None):
-        if not ledger_values:
+    def save_ledger_values(ledger_values, stock_result=None):
+        if not ledger_values and not (stock_result and stock_result.cases_with_deprecated_transactions):
             return
 
-        deprecated_form_id = deprecated_form.orig_id if deprecated_form else None
-
-        for ledger_value in ledger_values:
-            transactions_to_save = ledger_value.get_live_tracked_models(LedgerTransaction)
-
-            with get_cursor(LedgerValue) as cursor:
-                try:
-                    cursor.execute(
-                        "SELECT save_ledger_values(%s, %s::{}, %s::{}[], %s)".format(
-                            LedgerValue_DB_TABLE,
-                            LedgerTransaction_DB_TABLE
-                        ),
-                        [ledger_value.case_id, ledger_value, transactions_to_save, deprecated_form_id]
-                    )
-                except InternalError as e:
-                    raise LedgerSaveError(e)
-
-            ledger_value.clear_tracked_models()
+        try:
+            if stock_result and stock_result.cases_with_deprecated_transactions:
+                db_cases = split_list_by_db_partition(stock_result.cases_with_deprecated_transactions)
+                for db_name, case_ids in db_cases.items():
+                    LedgerTransaction.objects.using(db_name).filter(
+                        case_id__in=case_ids,
+                        form_id=stock_result.xform.form_id
+                    ).delete()
+
+            for ledger_value in ledger_values:
+                db_name = get_db_alias_for_partitioned_doc(ledger_value.case_id)
+                transactions_to_save = ledger_value.get_live_tracked_models(LedgerTransaction)
+
+                with transaction.atomic(using=db_name, savepoint=False):
```

Comment:
These can get called from outside of `save_processed_models` so still good to have these here.